semantic_release/changelog/changelog.py

```diff
@@ -1,8 +1,32 @@
+import re
+
 from typing import Iterable, Optional
 
 from ..settings import config
 
 
+def add_pr_link(owner: str, repo_name: str, message: str) -> str:
+    """
+    GitHub release notes automagically link to the PR, but changelog markdown
+    doesn't. Replace (#123) at the end of a message with a markdown link.
+    """
+
+    pr_pattern = re.compile(r'\s+\(#(\d{1,8})\)$')
+    match = re.search(pr_pattern, message)
+
+    if match:
+        pr_number = match.group(1)
+        url = (
+            f"https://gitlab.com/{owner}/{repo_name}/-/merge_requests/{pr_number}"
+            if config.get("hvcs") == "gitlab"
```

Comment:
Could you add a test for GitLab links?

Reply:
Sure!
Is a unit test for `add_github_link()` good (added), or do you also want me to add a test for `test_markdown_changelog()` / `test_changelog_table()` that sets hvcs to gitlab? (note: there is still no test coverage for the one that does the same thing for commit hashes).

I don't use GitLab much, but did verify that this _should_ be the right link pattern, e.g.,
https://gitlab.com/muttmua/mutt/-/merge_requests/120

Reply:
Update: added a separate test for gitlab `test_markdown_changelog()` as well as the unit test; that will also test `get_hash_link()` indirectly.